components/prism-typoscript.js

```diff
@@ -0,0 +1,31 @@
+    }, {
+        pattern: /pagets__.*(?:\s|=|.)/g
+    }],
```

Comment:
1. Any grammar token `{ pattern: /foo/ }` can be replaced with `/foo/`. Please also do this with all other objects of this form.
1. Please sort long keyword lists. This makes it easy to identify missing or duplicate keywords.
1. Regarding `/pagets__.*(?:\s|=|.)/`
    1. Shouldn't it start with `\b`?
    1. `.*` is greedy. Does that mean it's allowed to match e.g. spaces?

```suggestion
	'tag': [
		/\b(?:_CSS_DEFAULT_STYLE|_DEFAULT_PI_VARS|_LOCAL_LANG|_offset|absRefPrefix|accessibility|accessKey|ACT|ACTIFSUB|ACTIFSUBR|ACTRO|addAttributes|addChr10BetweenParagraphs|addClasses|addClassesCol|addClassesImage|addExtUrlsAndShortCuts|addItems|additionalHeaders|additionalParams|addParams|addQueryString|adjustItemsH|adjustSubItemsH|adminPanelStyles|admPanel|after|afterImg|afterImgLink|afterImgTagParams|afterROImg|afterWrap|age|alertPopups|align|all|allImageCaptions|allow|allowCaching|allowedAttribs|allowedClasses|allowedCols|allowedNewTables|allowTags|allowTVlisting|allSaveFunctions|allStdWrap|allWrap|alt_print|alternateBgColors|alternativeSortingField|alternativeTempPath|altImgResource|altLabels|altTarget|altText|altUrl|altUrl_noDefaultParams|altWrap|always|alwaysActivePIDlist|alwaysLink|alwaysShowClickMenuInTopFrame|andWhere|angle|antiAlias|append|applicationContext|applyTotalH|applyTotalW|archive|archiveTypoLink|arrayReturnMode|arrowACT|arrowImgParams|arrowNO|as|ascii|ATagAfterWrap|ATagBeforeWrap|ATagParams|ATagTitle|atLeast|atMost|attribute|auth|autoLevels|autonumber|backColor|backend_layout|BackendLayouts|background|badMess|baseURL|BE|be_groups|be_users|before|beforeImg|beforeImgLink|beforeImgTagParams|beforeLastTag|beforeROImg|beforeWrap|begin|beLoginLinkIPList|beLoginLinkIPList_login|beLoginLinkIPList_logout|bgCol|bgImg|blankStrEqFalse|blur|bm|bodyTag|bodyTagAdd|bodyTagCObject|bodyTagMargins|bodytext|border|borderClass|borderCol|borderPadding|borderSpace|bordersWithin|borderThick|borderWidth|bottomBackColor|bottomContent|bottomHeight|bottomImg|bottomImg_mask|BOX|br|browse|browser|brTag|bullet|bulletlist|bullets|bytes|cache|cache_clearAtMidnight|cache_period|caption|caption_stdWrap|captionAlign|captionHeader|captionSplit|CARRAY|case|casesensitiveComp|categories|cellpadding|cellspacing|centerImgACT|centerImgCUR|centerImgNO|centerLeftImgACT|centerLeftImgCUR|centerLeftImgNO|centerRightImgACT|centerRightImgCUR|centerRightImgNO|char|charcoal|charMapConfig|check|CHECK|class|classesAnchor|classesCharacter|classesImage|classesParagraph|classicPageEditMode|classStdWrap|clear|clearCache|clearCache_disable|clearCache_pageGrandParent|clearCache_pageSiblingChildren|clearCacheCmd|clearCacheLevels|clearCacheOfPages|CLEARGIF|clickMenuTimeOut|clickTitleMode|clipboardNumberPads|cMargins|cObj|cObject|cObjNum|colCount|collapse|collections|color|color1|color2|color3|color4|colors|colour|colPos|colPos_list|colRelations|cols|colSpace|colspan|columns|COLUMNS|columnSpacing|columnStdWrap|COMMENT|comment_auto|commentWrap|compensateFieldWidth|compressCss|compressJs|compX|compY|concatenateCss|concatenateJs|conf|config|CONFIG|constant|constants|CONSTANTS|content|content_from_pid_allowOutsideDomain|contextMenu|copy|copyLevels|count_HMENU_MENUOBJ|count_menuItems|count_MENUOBJ|create|createFoldersInEB|crop|csConv|CSS_inlineStyle|CTABLE|CType|CUR|CURIFSUB|CURIFSUBRO|current|CURRO|curUid|cut|cWidth|data|dataArray|dataKey|dataProcessing|dataWrap|date|date_stdWrap|datePrefix|dayofmonth|dayofweek|DB|db_list|debug|debugData|debugFunc|debugItemConf|debugRenumberedObject|default|defaultAlign|defaultCmd|defaultFileUploads|defaultHeaderType|defaultOutput|defaults|defaultType|delete|denyTags|depth|DESC|description|device|dimensions|directImageLink|directionLeft|directionUp|directory|directReturn|disableAdvanced|disableAllHeaderCode|disableAltText|disableBigButtons|disableBodyTag|disableCacheSelector|disableCharsetHeader|disabled|disableDelete|disableDocSelector|disableHideAtCopy|disableIconLinkToContextmenu|disableItems|disableNewContentElementWizard|disableNoMatchingValueElement|disablePageExternalUrl|disablePrefixComment|disablePrependAtCopy|disableSearchBox|disableSingleTableView|disableTabInTextarea|displayActiveOnLoad|displayContent|displayFieldIcons|displayIcons|displayMessages|displayQueries|displayRecord|displayTimes|distributeX|distributeY|div|DIV|doctype|doctypeSwitch|DOCUMENT_BODY|doktype|doNotLinkIt|doNotShowLink|doNotStripHTML|dontCheckPid|dontFollowMouse|dontHideOnMouseUp|dontLinkIfSubmenu|dontShowPalettesOnFocusInAB|dontWrapInTable|doubleBrTag|dWorkArea|dynCSS|edge|edit|edit_access|edit_docModuleUplaod|edit_docModuleUpload|edit_pageheader|edit_RTE|editFieldsAtATime|editFormsOnPage|editIcons|editIconsStdWrap|editNoPopup|editPanel|EFFECT|element|elements|else|email|emailMeAtLogin|emailMess|emboss|emptyTitleHandling|enable|encapsLines|encapsLinesStdWrap|encapsTagList|end|entryLevel|equalH|equals|everybody|excludeDoktypes|excludeUidList|expAll|expand|explode|ext|external|externalBlocks|extTarget|face|fallbackRendering|false|FE|fe_adminLib|fe_groups|fe_users|feadmin|field|fieldDelimiter|fieldName|fieldOrder|fieldRequired|fields|fieldWrap|file|file1|file2|file3|file4|file5|filelink|filelist|fileTarget|firstLabel|firstLabelGeneral|fixAttrib|flip|flop|fluidtemplate|folder|folders|folderTree|foldoutMenu|foldSpeed|foldTimer|fontColor|fontFile|fontOffset|fontSize|fontSizeMultiplicator|fontTag|forceDisplayFieldIcons|forceDisplayIcons|forceTemplateParsing|forceTypeValue|FORM|format|frame|FRAME|frameReloadIfNotInFrameset|frameSet|FRAMESET|freezeMouseover|ftu|function|Functions|gamma|gapBgCol|gapLineCol|gapLineThickness|gapWidth|get|getBorder|getLeft|getRight|global|globalNesting|globalString|globalVar|gmenu_foldout|GMENU_FOLDOUT|gmenu_layers|GMENU_LAYERS|goodMess|GP|gray|group|groupBy|groupid|header|header_layout|headerComment|headerData|headerSpace|headTag|height|helpText|hidden|hiddenFields|hide|hideButCreateMap|hideMenuTimer|hideMenuWhenNotOver|hidePStyleItems|hideRecords|hideSubmoduleIcons|highColor|history|hostname|hour|html|HTML|HTMLparser|HTMLparser_tags|htmlSpecialChars|htmlTag_dir|htmlTag_langKey|htmlTag_setParams|http|icon|icon_image_ext_list|icon_link|iconCObject|iconTitle|id|IENV|if|ifBlank|ifEmpty|IFSUB|IFSUBRO|image|image_compression|image_effects|image_frames|imageColumnStdWrap|imageLinkWrap|imagePath|images|imageStdWrap|imageStdWrapNoWidth|imageTextSplit|imageWrapIfAny|imgList|imgMap|imgMapExtras|imgMax|IMGMENUITEM|imgNameNotRandom|imgNamePrefix|imgObjNum|imgParams|imgPath|imgResource|imgStart|imgTagStdWrap|imgText|IMGTEXT|inBranch|inc|INCLUDE_TYPOSCRIPT|includeCSS|includeCSSLibs|includeJS|includeJSFooter|includeJSFooterlibs|includeJSLibs|includeLibrary|includeLibs|includeNotInMenu|includeRecordsWithoutDefaultTranslation|includeSpacer|incT3Lib_htmlmail|index|index_descrLgd|index_enable|index_externals|index_metatags|info|inlineStyle2TempFile|innerStdWrap|innerStdWrap_all|innerWrap|innerWrap2|input|inputLevels|insertClassesFromRTE|insertData|insertDmailerBoundaries|intensity|intTarget|intval|invert|IP|IProcFunc|isFalse|isGreaterThan|isInList|isLessThan|isPositive|isTrue|itemArrayProcFunc|itemH|items|itemsProcFunc|iterations|join|jsmenu|JSMENU|JSMENUITEM|JSwindow|JSWindow|JSwindow_params|jumpurl|jumpUrl|jumpurl_enable|jumpurl_mailto_disable|jumpUrl_transferSession|keep|keepEntries|keepNonMatchedTags|key|keyword3|label|LABEL|labelStdWrap|labelWrap|lang|language|language_alt|languageField|lastRowStdWrap|layer_menu_id|layerStyle|layout|layoutKey|layoutRootPath|layoutRootPaths|left|leftIcons|leftImgACT|leftImgCUR|leftImgNO|leftjoin|leftOffset|levels|leveltitle|leveluid|lib|limit|line|lineColor|lineThickness|linkParams|linkPrefix|linkTitleToSelf|linkVars|linkWrap|list|listNum|listOnlyInSingleTableView|LIT|lll|lm|locale_all|localNesting|locationData|lockFilePath|lockPosition|lockPosition_addSelf|lockPosition_adjust|lockToIP|login|loginUser|longdescURL|lowColor|lower|LR|mailform|mailto|main|mainScript|makelinks|markerWrap|marks|mask|max|maxAge|maxChars|maxGalleryWidth|maxGalleryWidthInText|maxH|maxHeight|maxItems|maxW|maxWidth|maxWInText|mayNotCreateEditShortcuts|media|mediaQuery|menu|menu_type|menuBackColor|menuHeight|menuName|menuOffset|menuWidth|message_page_is_being_generated|message_preview|meta|metaCharset|method|min|minH|minItems|minute|minW|mod|mode|module|moduleMenuCollapsable|month|move_wizard|MP_defaults|MP_disableTypolinkClosestMPvalue|MP_mapRootPoints|multimedia|MULTIMEDIA|name|navFrameResizable|navFrameWidth|negate|nesting|netprintApplicationLink|neverHideAtCopy|new|NEW|new_wizard|newPageWiz|newRecordFromTable|newWindow|newWizards|next|nicetext|niceText|NO|no_cache|no_search|noAttrib|noCache|noCaption|noCols|noCreateRecordsLink|noLink|noMatchingValue_label|noMenuMode|nonCachedSubst|none|nonTypoTagStdWrap|nonTypoTagUserFunc|nonWrappedTag|noOrderBy|noPageTitle|noResultObj|noRows|noRowsStdWrap|noScaleUp|noStretchAndMarginCells|noThumbsInEB|noThumbsInRTEimageSelect|notification_email_charset|notification_email_encoding|notification_email_urlmode|noTrimWrap|noValueInsert|numRows|obj|offset|offsetWrap|onlineWorkspaceInfo|onlyCurrentPid|opacity|options|order|orderBy|OTABLE|outerWrap|outline|outputLevels|override|overrideAttribs|overrideId|overridePageModule|overrideWithExtension|page|PAGE_TARGET|PAGE_TSCONFIG_ID|PAGE_TSCONFIG_IDLIST|PAGE_TSCONFIG_STR|pageFrameObj|pageGenScript|pages|pages_language_overlay|pageTitleFirst|pageTree|parameter|params|parseFunc|parseFunc_RTE|parser|parseTarget|partialRootPath|partialRootPaths|password|paste|path|permissions|perms|pid|pid_list|pidInList|PIDinRootline|PIDupinRootline|pixelDensity|pixelSpaceFontSizeRef|plaintextLib|plainTextStdWrap|plugin|postCObject|postform|postform_newThread|postLineBlanks|postLineChar|postLineLen|postUserFunc|postUserFuncInt|preBlanks|preCObject|prefix|prefixComment|prefixLocalAnchors|prefixRelPathWith|preIfEmptyListNum|preLineBlanks|preLineChar|preLineLen|prepend|preRenderRegisters|preserveEntities|preUserFunc|prev|preview|previewBorder|prevnextToSection|printheader|prioriCalc|proc|properties|protect|protectLvar|publish|publish_levels|QEisDefault|quality|radio|RADIO|radioWrap|range|rawUrlEncode|recipient|RECORDS|recursive|recursiveDelete|redirect|redirectToURL|reduceColors|references|register|relation|relativeToParentLayer|relativeToTriggerItem|relPathPrefix|remap|remapTag|REMOTE_ADDR|removeBadHTML|removeDefaultJS|removeEmptyEntries|removeIfEquals|removeIfFalse|removeItems|removeObjectsOfDummy|removePrependedNumbers|removeTags|removeWrapping|renderCharset|rendering|renderMethod|renderObj|renderWrap|replace|replacement|REQ|required|reset|resources|resultObj|returnKey|returnLast|returnUrl|rightImgACT|rightImgCUR|rightImgNO|rightjoin|rm|rmTagIfNoAttrib|RO|RO_chBgColor|rootline|rotate|rowCount|rows|rowSpace|rowStdWrap|RTE|RTE_compliant|RTEfullScreenWidth|rules|sample|saveClipboard|saveDocNew|search|SEARCHRESULT|secondRow|section|sectionIndex|select|select_key|selectFields|sendCacheHeaders|separateRows|separator|set|setContentToCurrent|setCurrent|setfixed|setFixedHeight|setFixedWidth|setJS_mouseOver|setJS_openPic|setOnly|setup|shadow|SHARED|sharpen|shear|short|shortcut|shortcut_onEditId_dontSetPageTree|shortcut_onEditId_keepExistingExpanded|shortcutFrame|shortcutGroups|shortcutIcon|show|showAccessRestrictedPages|showActive|showClipControlPanelsDespiteOfCMlayers|showFirst|showHiddenPages|showHiddenRecords|showHistory|showPageIdWithTitle|showTagFreeClasses|simulateDate|simulateUserGroup|singleCaption|singleNoCaption|singlePid|singleStdWrap|site_author|site_reserved|sitemap|sitetitle|siteUrl|size|small|smallRetina|solarize|sorting|source|sourceCollection|space|spaceAfter|spaceBefore|spaceBelowAbove|spaceLeft|spaceRight|spacing|spamProtectEmailAddresses|spamProtectEmailAddresses_atSubst|spamProtectEmailAddresses_lastDotSubst|SPC|special|split|splitCaption|splitChar|splitRendering|src|srcsetCandidate|startInTaskCenter|stayFolded|stdheader|stdWrap|stdWrap2|strftime|stripHtml|styles|stylesheet|submenuObjSuffixes|subMenuOffset|submit|subparts|subst_elementUid|substMarksSeparately|substring|swirl|sword|sword_noMixedCase|SWORD_PARAMS|sword_standAlone|sys_dmail|sys_domain|sys_filemounts|sys_language_mode|sys_language_overlay|sys_language_softMergeIfNotBlank|sys_language_uid|sys_note|sys_template|system|tabel|table|tableCellColor|tableParams|tables|tableStdWrap|tableStyle|tableWidth|tags|target|TCAdefaults|TCEFORM|TCEMAIN|TDparams|temp|template|templateContent|templateFile|templateName|templateRootPath|templateRootPaths|text|textarea|textMargin|textMargin_outOfText|textMaxLength|textmedia|textObjNum|textpic|textPos|textStyle|thickness|this|thumbnailsByDefault|tile|time_stdWrap|tipafriendLib|title|titleInLink|titleInLinkAndImg|titleLen|titleTagFunction|titleText|tm|TMENU_LAYERS|TMENUITEM|token|top|topOffset|totalWidth|transparentBackground|transparentColor|treatIdAsReference|treeLevel|trim|true|tsdebug|tsdebug_tree|TSFE|tt_content|twice|type|typeNum|types|typolink|typolinkCheckRootline|uid|uidInList|uniqueGlobal|uniqueLocal|unset|unsetEmpty|updated|uploadFieldsInTopOfEB|uploads|upper|url|us|useCacheHash|useLargestItemX|useLargestItemY|user|user_task|useragent|USERDEF1|USERDEF1RO|USERDEF2|USERDEF2RO|userdefined|useRegExp|userFunc|userfunction|usergroup|userid|USERNAME_substToken|userProc|USR|USRRO|value|valueArray|version|view|wave|web_func|web_info|web_layout|web_list|web_t|where|width|wiz|wordSpacing|workArea|workOnSubpart|wrap|wrap1|wrap2|wrap3|wrapAfterTags|wrapAlign|wrapFieldName|wrapItemAndSub|wrapNonWrappedLines|wraps|xhtml_cleanig|xhtml_strict|xhtml_trans|xmlprologue|xPosOffset|XY|ypMenu|yPosOffset)\b/,
		/pagets__.*(?:\s|=|.)/,
	],
```

Reply:
About 1.: Sure, it's just more readably (imho), if there are more groups. But jey, sure.
About 2: I've managed to remove the long list completely. I add this update asap.
About 3: This was removed too in the update, so nevermind. ;)

Reply:
So about 1. again: Is this an issue for you, or is it okay to use blocks like this below. I found it just way more readable.

```JS
'baz': [
    {
        pattern: /foo/,
    },
    {
        pattern: /bar/,
        lookbehind: true,
        greedy: true
    }
],
```

Reply:
> So about 1. again:

That's acceptable. I mainly don't want those two:

```js
'foo': {
    pattern: /foo/
},
'bar': [ // all items of this form
    {
        pattern: /bar/
    },
    {
        pattern: /baz/
    }
]
```

instead of simply:

```js
'foo': /foo/,
'bar': [
    /bar/,
    /baz/
]
```

I get that a random regex literal in an array object looks a little strange, so it's ok to use the object-form there.